best-practices/client-side/framework/react/components/media-cropper/src/components/media-cropper/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react';
+
+import ImageRenderer from '../image-renderer';
+
+export default class MediaCropper extends Component {
+    constructor(props) {
+        super(props);
+        const media = new Image();
+        media.src = `https://via.placeholder.com/${props.width}x${props.height}.png?text=click%20to%20add`;
+
+        this.state = {
+            zoom: 1,
+            transX: 0,
+            transY: 0,
+            media,
+            isVideo: false,
+            dragging: false,
+            mouseStart: {
+                x: 0,
+                y: 0
+            },
+            moved: false
+        };
+
+        this.click = this.click.bind(this);
+        this.mouseDown = this.mouseDown.bind(this);
+        this.mouseMove = this.mouseMove.bind(this);
+        this.mouseUp = this.mouseUp.bind(this);
+        this.mouseWheel = this.mouseWheel.bind(this);
+        this.loadFile = this.loadFile.bind(this);
+    }
+
+    componentDidUpdate(props, state) {
+        if (this.state.dragging && !state.dragging) {
+            document.addEventListener('mousemove', this.mouseMove);
+            document.addEventListener('mouseup', this.mouseUp);
+            document.addEventListener('mouseleave', this.mouseUp);
+        } else if (!this.state.dragging && state.dragging) {
+            document.removeEventListener('mousemove', this.mouseMove);
+            document.removeEventListener('mouseup', this.mouseUp);
+            document.removeEventListener('mouseleave', this.mouseUp);
+        }
```

Comment:
How to do feel about moving these to `useEvent` or `useSelector` react components for detecting component updates?

Reply:
Sorry, not really experienced with React, what are those?

Reply:
No problem, Im sorta new too!
Those are all part of https://react-redux.js.org

Specifically Hooks: https://react-redux.js.org/api/hooks

I think they are neat.

Reply:
oh, gotcha. This component will not be using redux as its state should be self-contained. Is there a particular reason you feel that would be useful here?

Reply:
A good reason for using hooks, specifically (those are _not_ redux-specific, but rather React core) is that they are the way that modern React handles local state.